### DISCUSSION TITLE
Passing second argument to function

### DISCUSSION THREAD

**Discussion by juankstylez** [nutgram/nutgram#361] [General]
I need to pass a second argument from

$this->next('secondStep');

toward

the next function of the Steps.

how do i do it try but it doesn't work.

$bot->sendMessage('First and Last Name');
         $this->next('askEmail', $Trimmedreferrer);
     }

     public function askEmail(Nutgram $bot, $ref)
     {
         // if not a callback query, ask again!
         if (!$bot->message()->text) {
             $this->askNameLastName($bot, $ref);
             return;
         }

$ref

mistake

ArgumentCountError: Too few arguments to function App\Http\Controllers\AuthController::askEmail(), 1 passed in D:\laragon\www\bot\vendor\nutgram\nutgram\src\Conversations\Conversation.php on line 110 and exactly 2 expected in D:\laragon\www\bot\app\Http\Controllers\AuthController.php:61

**Comment by Lukasss93** [comment]
Read this: https://github.com/nutgram/nutgram/discussions/240

PS: next time search on documentation/discussions

**Reply by juankstylez** [reply]
Brou ! 
I already tried that, my problem is when it does not receive a text and it is returned to first step it generates the error.

How do I return the $ref back to the first step?

thanksss ! I have read a lot of your documentation, I love nutgram

// if is not a callback query, ask again!
        if (!$bot->message()->text) {
            $this->askNameLastName($bot);
            return;
        }

**Reply by juankstylez** [reply]
It should be like that ? because it doesn't work for me either.

// if is not a callback query, ask again!
if (!$bot->message()->text) {
$this->askNameLastName($bot, $ref);
return;
}

**Reply by Lukasss93** [reply]
just use a class property as described in the last link I sent you
